client/src/components/ResultConditions.js

```diff
@@ -0,0 +1,188 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { inject } from 'lib/Injector';
+import { Row, Col } from 'reactstrap';
+import fieldHolder from 'components/FieldHolder/FieldHolder';
+import i18n from 'i18n';
+
+/**
+ * Result conditions allow a CMS user to set one condition for displaying the column if the contents
+ * of it match or do not match a given string.
+ */
+class ResultConditions extends Component {
+  constructor(props) {
+    super(props);
+
+    const value = props.value || {};
+
+    // Set initial state values
+    this.state = {
+      [this.getFieldName('match-select', props)]: value['match-select'] || 1,
+      [this.getFieldName('match-text', props)]: value['match-text'] || '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  /**
+   * Returns the namespaced form field name for the given field
+   *
+   * @param {string} fieldName
+   * @param {object} props If provided, will be used instead of this.props
+   * @returns {string}
+   */
+  getFieldName(fieldName, props = {}) {
+    const name = props.name || this.props.name;
+    return `${name}-${fieldName}`;
+  }
+
+  /**
+   * Gets the current value of the "must match" select
+   *
+   * @returns {string}
+   */
+  getSelectValue() {
+    return this.state[this.getFieldName('match-select')];
+  }
+
+  /**
+   * Gets the current value of the filter condition text input
+   *
+   * @returns {string}
+   */
+  getInputValue() {
+    return this.state[this.getFieldName('match-text')];
+  }
+
+  /**
+   * Gets a copy of the state, used to store in the hidden input field
+   *
+   * @returns {object}
+   */
+  getValue() {
+    return {
+      'match-select': this.getSelectValue(),
+      'match-text': this.getInputValue(),
+    };
```

Comment:
What do you think about returning an array with just a single object - just future-proofing for the eventual multi-filter component that was in the designs?

Reply:
Good thinking. I'll also remove the `getMatchCondition()` and `getMatchText()` methods from the form field, since they won't be relevant once there are multiple

Reply:
I tried updating this but it was a rabbit hole. At some point there'll have to be a bunch more work to make it work with multiple options but I don't want to go to that effort now.